pkg/gofr/http/router.go

```diff
@@ -13,23 +13,30 @@ import (
 // Router is responsible for routing HTTP request.
 type Router struct {
 	mux.Router
+	RegisteredRoutes *[]string
 }
 
 type Middleware func(handler http.Handler) http.Handler
 
 // NewRouter creates a new Router instance.
-func NewRouter(c *container.Container) *Router {
+func NewRouter(c *container.Container, middlewareConfigs map[string]string) *Router {
```

Comment:
this change can be avoided as we are changing function signature of an exported function!

What if we move this logic to httpServer.go, i.e registering middlewares.

Reply:
Sure will shift the logic there